### ISSUE TITLE
 Need type check and unified error message for search partition name

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#6592]
#### Describe the bug:
Error message of search partition is a little confused
1 <class 'int'>: 1
Value must be iterable
2 <class 'list'>: [1, '2', 3]
TypeError: 1 has type int, but expected one of: bytes, unicode
3 <class 'tuple'>: (1,)
TypeError: 1 has type int, but expected one of: bytes, unicode
4  <class 'dict': {1: 1}
TypeError: 1 has type int, but expected one of: bytes, unicode
5  <class 'str'>: 12 s
message=Invalid partition name: . Partition name should not be empty.
6  <class 'str'>: (mn)
 invalid partition names
7 <class 'str'>: %$#
(code=1, message=PartitonName: % not found)

#### Expected behavior:
(1) for unsupported type, can we report one illegal message?
(2) for supported type, report current messge: PartitionName: ** not found

#### Steps/Code to reproduce:
```python
from pymilvus_orm import CollectionSchema, FieldSchema
from pymilvus_orm import Collection
from pymilvus_orm import connections
from pymilvus_orm.types import DataType
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=128)
schema = CollectionSchema(fields=[int64_field, float_field, float_vector])
connections.connect()
import numpy as np
collection = Collection("test_search_collection_binbin_6", schema=schema)
dim = 128
nb = 10000
nq = 2
limit = 10
import random
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], vectors])
collection.load()
res = collection.search({'a': 1}, "float_vector", default_search_params, limit, "int64 >= 0", 1)
```

#### Environment:
- Milvus version(e.g. v1.1.0 or 8b23a93): 
- SDK version(e.g. pymilvus v2.0.0rc2):
Name: pymilvus
Version: 2.0.0rc3.dev4
Name: pymilvus-orm
Version: 2.0.0rc3.dev5

- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:

**Comment by binbinlv** [comment]
And same issue on "output_fields" part, could you please modify it together?

**Comment by DragonDriver** [comment]
thx for pointing out this. it will be fixed by [pr #613 of pymilvus](https://github.com/milvus-io/pymilvus/pull/613). pls retest this after it was merged.

/unassign
/assign @binbinlv

**Comment by binbinlv** [comment]
> thx for pointing out this. it will be fixed by [pr #613 of pymilvus](https://github.com/milvus-io/pymilvus/pull/613). pls retest this after it was merged.
> 
> /unassign
> /assign @binbinlv

OK, thanks, I will verified it after the code merge.

**Comment by binbinlv** [comment]
Verified, and fixed.

Name: pymilvus
Version: 2.0.0rc3.dev6
Name: pymilvus-orm
Version: 2.0.0rc3.dev6

Result:
`partition_name_array` value (1,) is illegal
